tools/socktap/router_context.cpp

```diff
@@ -24,10 +24,10 @@ RouterContext::RouterContext(raw_protocol::socket& socket, const geonet::MIB& mi
     router_.set_address(mib_.itsGnLocalGnAddr);
     router_.set_access_interface(request_interface_.get());
     router_.set_transport_handler(geonet::UpperProtocol::BTP_B, &dispatcher_);
-    update_position_vector();
+    router_.set_security_entity(&security_entity);
 
+    update_position_vector();
     do_receive();
-    trigger_.schedule();
```

Comment:
Why do you remove the scheduling of the time trigger?

Reply:
I initially removed it, because I introduced a setter for the security entity. If the security entity isn't set, but `MIB::itsGnSecurity` is true, that one causes an error because of an unset security entity, which I solved by removing it and deferring the `update_position_vector()` by 0 seconds, which allows setting the security entity right after the constructor call.

In the squashed commit I went with setting the security entity always (hence it's in the constructor), as otherwise no signed packets can be received.

I left it removed, because it's anyway directly called in `update_position_vector()`.

I can of course move that to a separate commit or remove it entirely from the commit.

Reply:
Okay, I see. Removed it from your commit.